### PR TITLE
COMPASS-4878: Add telemetry events v2

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

Add telemetry events for the time-series epic.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-query-bar/src/stores/query-bar-store.js [hunk 1]
```diff
@@ -753,8 +755,9 @@ const configureStore = (options = {}) => {
   }
 
   if (options.namespace) {
-    store.onCollectionChanged(options.namespace);
+    store.onCollectionChanged(options.namespace, options.isTimeSeries);
```
Comment (mcasimir) [inline packages/compass-query-bar/src/stores/query-bar-store.js:758]: Where is this coming from?

File: packages/compass-metrics/src/modules/features.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ const CollectionResource = FeatureResource.extend({
   eventTrackers: ['stitch'],
   fetched: function(metadata, callback) {
     this._send_event(metadata, callback);
+  },
+  created: function(metadata, callback) {
```
Comment (mcasimir) [inline packages/compass-metrics/src/modules/features.js:48]: why is this necessary?

**Review by mcasimir — APPROVED** [review]
👍

File: packages/compass-metrics/src/modules/features.js (hunk 2 above)
Comment (alenakhineika) [reply packages/compass-metrics/src/modules/features.js:48]: This is how the telemetry module works, you set rules like:

```
{
    registryEvent: 'compass:collection:created',
    resource: 'Collection',
    action: 'created',
    condition: () => true,
    metadata: (version, state) => ({ ... })
  }
```

and then based on resource and action it finds a method in feature.js and checks where to send telemetry based on `eventTrackers` property.

File: packages/compass-query-bar/src/stores/query-bar-store.js (hunk 1 above)
Comment (alenakhineika) [reply packages/compass-query-bar/src/stores/query-bar-store.js:758]: It is already in the store and is being set when a new collection is selected.
